[18.0][MIG] base_location: Migration to 18.0

Kimkhoi3010 opened the pull request.

nbessi committed:
[MV] rename better_zip to base_location

Commit:
base_location: [IMP] Spanish translation and translation template.

nbessi committed:
[FIX] mutable default in function signature

Commit:
[FIX] base_location: Changed partner view to avoid extrange behaviour when company country is filled.

Commit:
[FIX] Add context propagation to base_location

Commit:
[Add] Add italian translation, change field class to edit-only for better-zip field

Commit:
base_location: Don't position zip_id after street2, to avoid layout issues with module partner_address_street3

hhgabelgaard committed:
Danish translation

Commit:
[IMP] base_location: Updated to v8 syntax and uses. Added filters in some views. Added two columns in respective tree views.

pedrobaeza committed:
[IMP] base_location_geonames_import: Several improvements and added hooks
* Added Icon.
* Improve module description and extracted to README.rst.
* Pass country instead of country_id for advance comparisons.
* Allow to transform city name.
* Some code style.
* Do not remove all entries of a country, but only not found.
* Include hooks for transforming some things.
* Include spanish translation.

Commit:
Add demo data

Brussels Belgium city

jjscarafia committed:
IMP Create new computed stored field 'display_name' that use zip, city, state and country in order to allow search using '%' from the m2o widget

hhgabelgaard committed:
Translated to Danish

yvaucher committed:
[MIG] base_location: Migration to 9.0

lmignon committed:
[IMP] Rename partner_identifiers -> partner_identification

lmignon committed:
partner_identification: several fix/imp

* [IMP] improve module description
* [IMP] Remove useless comments
* [FIX] Complete incomplete sentence
* [IMP] Replace field 'state' by 'status' in res_partner_id_number
* [IMP] Add new field 'Place of Issuance'
* [FIX] Readme formatting
* [IMP] status is now a selection field
* [IMP] use method to provide the default value for validation_code
* [IMP] Add help texts
* [FIX] Add missing constrains on category_id in res_partner.id_number
        The number must be validated also when we change the category

oca-transbot committed:
OCA Transbot updated translations from Transifex

lmignon committed:
[FIX] partner_identification: prevent integrity error on delete ID

oca-transbot committed:
OCA Transbot updated translations from Transifex

OpenCode committed:
[MIG] base_location: Migrated to 10.0

* Headers shortened
* Move cities management to settings

jjscarafia committed:
[ADD] partner_identification: sales_team dep

jjscarafia committed:
[FIX] partner_identification: pylint

kitcharoenp committed:
[FIX][10.0] base_location :  "better_zip_ids" field set colspan="2" in state view

lasley committed:
[IMP] partner_identification: Add context override (#373)

Allow for context override of validations using ``id_no_validate``

lasley committed:
[10.0][IMP] base_location(geonames): Add Lat/Long (#405)

* [IMP] base_location: Add lat & long to `better.zip`
* Add latitude and longitude columns to `better.zip`

* [IMP] base_location_geonames_import: Add lat/long
* Add support for latitude & longitude to genomes importer

lasley committed:
[10.0][IMP] partner_identification: Add field computation and inverses (#419)

* [IMP] partner_identification: Add field computation and inverses
* Add methods to allow for computation and inverse of an ID field of a specific category type

* [IMP] partner_identification: Add search option

lasley committed:
[FIX] partner_identification: Infinite loop in search (#436)

etobella committed:
[MIG] base_location

oca-transbot committed:
OCA Transbot updated translations from Transifex

simahawk committed:
[11.0][MIG] partner_identification

simahawk committed:
partner_identification: fix/imp tests on res.partner

pedrobaeza committed:
[IMP] base_location: Include onchange for state

Incredibly not included in Odoo core.

WohthaN committed:
[IMP] base_location: name_search improvement (#585)

oca-travis committed:
[UPD] Update partner_identification.pot

etobella committed:
Translated using Weblate (Spanish)

Currently translated at 66.7% (24 of 36 strings)

Translation: partner-contact-11.0/partner-contact-11.0-base_location
Translate-URL: https://translation.odoo-community.org/projects/partner-contact-11-0/partner-contact-11-0-base_location/es/

rschnapka committed:
Translated using Weblate (German)

Currently translated at 100,0% (36 of 36 strings)

Translation: partner-contact-11.0/partner-contact-11.0-base_location
Translate-URL: https://translation.odoo-community.org/projects/partner-contact-11-0/partner-contact-11-0-base_location/de/

aitorbouzas committed:
[MIG] base_location: Migration to 12.0

This module has now been refactored to be more consistent with what base_address_city offers to the location management.
Added dependency to contacts so that I could change the menu location for cities / zip management.

Now, every res.city record has a relation One2many to res.city.zip (old res.better.zip). This way, every zip has a realted city too.
Zips can be searched through city code, zip or city name (same as before).

Modified tests and deleted not needed tests.

Added sql contraints so that zips and cities are unique within it's country / state / city.

pedrobaeza committed:
[IMP] base_location: Add migration scripts

tarteo committed:
[MIG] partner_identification: Migration to 12.0

[FIX] Tests

[FIX] data sequence

Fix openerp to odoo in comment

[REM] at_install, post_install

[ADD] Model descriptions

[REM] Extra spaces

OCA-git-bot committed:
[UPD] README.rst

oca-travis committed:
[UPD] Update partner_identification.pot

primes2h committed:
Translated using Weblate (Italian)

Currently translated at 97.9% (46 of 47 strings)

Translation: partner-contact-12.0/partner-contact-12.0-partner_identification
Translate-URL: https://translation.odoo-community.org/projects/partner-contact-12-0/partner-contact-12-0-partner_identification/it/

oca-transbot committed:
Update translation files

Updated by Update PO files to match POT (msgmerge) hook in Weblate.

oca-travis committed:
[UPD] Update partner_identification.pot

oca-transbot committed:
Update translation files

Updated by "Update PO files to match POT (msgmerge)" hook in Weblate.

Translation: partner-contact-12.0/partner-contact-12.0-partner_identification
Translate-URL: https://translation.odoo-community.org/projects/partner-contact-12-0/partner-contact-12-0-partner_identification/

mvrodriguez committed:
Translated using Weblate (Spanish)

Currently translated at 97.9% (46 of 47 strings)

Translation: partner-contact-12.0/partner-contact-12.0-partner_identification
Translate-URL: https://translation.odoo-community.org/projects/partner-contact-12-0/partner-contact-12-0-partner_identification/es/

eLBati committed:
FIX base_location blocking editing company address.

Steps:

 - Open company form
 - Set 'city completion' field

Get 'The state of the partner My Company differs from that in location X'

Disabling _check_zip while writing 'zip' fields from company, as incompatible with the sequence of write operations.

Automatic test is added too.

OCA-git-bot committed:
[ADD] icon.png

OCA-git-bot committed:
[UPD] README.rst

josepmy committed:
Translated using Weblate (Spanish)

Currently translated at 57.6% (19 of 33 strings)

Translation: partner-contact-12.0/partner-contact-12.0-base_location
Translate-URL: https://translation.odoo-community.org/projects/partner-contact-12-0/partner-contact-12-0-base_location/es/

pedrobaeza committed:
[FIX+MIG] base_location: 3 things:

* Don't need cities for migrating

  Previous script required to have cities populated for working, but that's not
  usual thing, as people in v11 may choose to not import them, and for older databases
  they even weren't that option.

  With this improve script, now we populate city table for ZIP entries without city,
  so the rest of the queries are properly executed in any case.

* Cover case of res_better_zip w/o country_id

* Avoid error on null zip numbers

pedrobaeza committed:
[MIG] base_location: Migration to 13.0

* Standard procedure
* Change v13 specifics
* Adapt tests + correct some practices

OCA-git-bot committed:
pre-commit, black, isort

pedrobaeza committed:
[FIX] base_location: Don't apply domain when cities are not enforced

And also reset city_id for avoiding incoherent information in that case.

Fixes #762

mileo committed:
Translated using Weblate (Portuguese (Brazil))

Currently translated at 100.0% (33 of 33 strings)

Translation: partner-contact-13.0/partner-contact-13.0-base_location
Translate-URL: https://translation.odoo-community.org/projects/partner-contact-13-0/partner-contact-13-0-base_location/pt_BR/

Commit:
[REF] partner_identification: Black python code

Commit:
[MIG] partner_identification: Migration to 13.0

oca-transbot committed:
Update translation files

Updated by "Update PO files to match POT (msgmerge)" hook in Weblate.

Translation: partner-contact-13.0/partner-contact-13.0-base_location
Translate-URL: https://translation.odoo-community.org/projects/partner-contact-13-0/partner-contact-13-0-base_location/

ntsirintanis committed:
[FIX] make _compute_identification() to assign a value no matter what

oca-travis committed:
[UPD] Update partner_identification.pot

OCA-git-bot committed:
[UPD] README.rst

oca-transbot committed:
Update translation files

Updated by "Update PO files to match POT (msgmerge)" hook in Weblate.

Translation: partner-contact-13.0/partner-contact-13.0-partner_identification
Translate-URL: https://translation.odoo-community.org/projects/partner-contact-13-0/partner-contact-13-0-partner_identification/

Oocademy committed:
Translated using Weblate (Dutch)

Currently translated at 93.9% (31 of 33 strings)

Translation: partner-contact-13.0/partner-contact-13.0-base_location
Translate-URL: https://translation.odoo-community.org/projects/partner-contact-13-0/partner-contact-13-0-base_location/nl/

OCA-git-bot committed:
pre-commit update

dong-z committed:
Translated using Weblate (Chinese (Simplified))

Currently translated at 100.0% (47 of 47 strings)

Translation: partner-contact-13.0/partner-contact-13.0-partner_identification
Translate-URL: https://translation.odoo-community.org/projects/partner-contact-13-0/partner-contact-13-0-partner_identification/zh_CN/

OCA-git-bot committed:
[UPD] README.rst

pedrobaeza committed:
[FIX+IMP] base_location: Re-import process

* Performance was very poor due to several reasons:

  * City name translatable.
  * Lack on indexes when deleting obsolete records.
  * Some prefetching.
* City was not being searched also by state, having the chance of returning
  incorrect matches if same city name on different states.
* Previous code was removing all previous records of zip entries!
* Remove possible obsolete FK keys after migrating from v11.

Commit:
Translated using Weblate (Portuguese (Brazil))

Currently translated at 100.0% (34 of 34 strings)

Translation: partner-contact-13.0/partner-contact-13.0-base_location
Translate-URL: https://translation.odoo-community.org/projects/partner-contact-13-0/partner-contact-13-0-base_location/pt_BR/

TDu committed:
[FIX][13.0] partner_identification id number check

OCA-git-bot committed:
partner_identification 13.0.1.0.1

ajaniszewska-dev committed:
[FIX][13.0] partner_identification: Add menu for Partner ID Numbers

pedrobaeza committed:
[IMP] : black, isort, prettier

oca-travis committed:
[UPD] Update partner_identification.pot

OCA-git-bot committed:
partner_identification 13.0.1.1.0

oca-transbot committed:
Update translation files

Updated by "Update PO files to match POT (msgmerge)" hook in Weblate.

Translation: partner-contact-13.0/partner-contact-13.0-partner_identification
Translate-URL: https://translation.odoo-community.org/projects/partner-contact-13-0/partner-contact-13-0-partner_identification/

pedrobaeza committed:
[MIG] base_location: Migration to 14.0

ajaniszewska-dev committed:
[13.0][IMP] Add missing field to ID Numbers tree view

OCA-git-bot committed:
partner_identification 13.0.1.2.0

pedrobaeza committed:
[IMP] base_location: Switch to computed writable fields

* Adapt code to this style
* Adapt tests and use Form for mimicking UI behavior

ypapouin committed:
[14.0][FIX] base_location: Update name when changing city's state or country

ibuioli committed:
Translated using Weblate (Spanish (Argentina))

Currently translated at 100.0% (47 of 47 strings)

Translation: partner-contact-13.0/partner-contact-13.0-partner_identification
Translate-URL: https://translation.odoo-community.org/projects/partner-contact-13-0/partner-contact-13-0-partner_identification/es_AR/

bosd committed:
Translated using Weblate (Dutch)

Currently translated at 21.3% (10 of 47 strings)

Translation: partner-contact-13.0/partner-contact-13.0-partner_identification
Translate-URL: https://translation.odoo-community.org/projects/partner-contact-13-0/partner-contact-13-0-partner_identification/nl/

carlosdauden committed:
[ADD] partner_pricelist_search: New module
TT27659

phuctranfxvn committed:
[IMP] partner_identification: black, isort, prettier

phuctranfxvn committed:
[MIG] partner_identification: Migration to 14.0

phuctranfxvn committed:
[IMP] partner_identification: Add python validation, improve view and fix default partner

phuctranfxvn committed:
[FIX] partner_identification: Do not validate number if validation code is not set

carlosdauden committed:
[MIG] partner_pricelist_search: Migration to v13.0
TT28442

pedrobaeza committed:
[IMP] partner_pricelist_search: black, isort, prettier

TDu committed:
[13.0][ADD] partner_identification_unique_by_category

This module extends the module `partner_identification` to enforce having
unique id numbers by categories.
This option can be set by category.

[UPD] Update partner_identification_unique_by_category.pot

[UPD] README.rst

[ADD] icon.png

braisab committed:
[MIG] partner_identification_unique_by_category: Migration to 14.0

pedrobaeza committed:
[FIX] base_location: Don't slow down the partner view

Using the computed allowed + domain with IN is very slow for loading
the partner view, more with the initial set, that gets the whole set
of records.

Luckily, the answer is the magic operator `=?`, that allows to express
a conditional domain that is not applied if the right leaf is False,
complying with the intention of the dynamic domain, and even expanding
it to more conditions for further assistance.

Fixes #1073

ivantodorovich committed:
[IMP] Check that the zip code matches the one of the zip_id

Also:
* Reorder checks by geo level: country, state, city, zip

ivantodorovich committed:
[IMP] Improve readability of with_context(skip_check_zip=True)

* Easier to read by splitting lines
* Also, set context before the loop

ivantodorovich committed:
[FIX] city_id no longer shown.

We want to use a domain that requires city_id to be on the view
but we can't add it directly there, otherwise _fields_view_get_address
in base_address_city won't do its magic, as it immediately returns
if city_id is already in there.

On the other hand, if city_id is not in the
views, odoo won't let us use it in zip_id's domain.

Fixes a bug introduced here: 5fd93fb8d138cb9a650410d49745b0777be43be0

ivantodorovich committed:
[IMP] Also empty zip_id when city_id or zip mismatch

Chanakya-OSI committed:
[14.0][IMP] Added the field 'color' to partner_id_category.

oca-travis committed:
[UPD] Update partner_identification_unique_by_category.pot

OCA-git-bot committed:
[UPD] README.rst

OCA-git-bot committed:
partner_identification_unique_by_category 14.0.1.0.1

victoralmau committed:
[MIG] partner_pricelist_search: Migration to 14.0

[UPD] Update partner_pricelist_search.pot

[UPD] README.rst

Update translation files

Updated by "Update PO files to match POT (msgmerge)" hook in Weblate.

Translation: partner-contact-14.0/partner-contact-14.0-partner_pricelist_search
Translate-URL: https://translation.odoo-community.org/projects/partner-contact-14-0/partner-contact-14-0-partner_pricelist_search/

cvinh committed:
[IMP][partner_identification] Allow attachments and chatter for res_partner_id_number model

ivantodorovich committed:
[FIX] base_location: address field sync zip_id

`_address_fields` are used by `update_address` [1], which is in turn
called by `_fields_sync` and `_children_sync`.

This mechanism is in charge of synchronizing parent/children address fields
for contacts of type == "contact", as they don't really have an address of
their own. They simply copy the address of their parent.

Without this fix, not only the zip_id is not sync'ed, but also in some cases
an exception may be raised by `_check_zip_id`, due to only some fields being
sync'ed but not all, creating inconsistencies between zip_id and other address
fields.

[1]: https://github.com/odoo/odoo/blob/20648ef21/odoo/addons/base/models/res_partner.py#L429

ivantodorovich committed:
[FIX] base_location: all users should be able to read

For some reason, this access was removed during the migration to 12.0 (67baf9196).
But this security is perfectly valid and aligned to Odoo core's security for `res.city`[^1].

Without this, you might get bitten while trying to implement website features that
require reading or setting partner's zip_id.

[^1]: https://github.com/odoo/odoo/blob/f5a5d9616/addons/base_address_city/security/ir.model.access.csv

educasilva committed:
Translated using Weblate (Portuguese (Portugal))

Currently translated at 100.0% (47 of 47 strings)

Translation: partner-contact-14.0/partner-contact-14.0-partner_identification
Translate-URL: https://translation.odoo-community.org/projects/partner-contact-14-0/partner-contact-14-0-partner_identification/pt_PT/

pedrobaeza committed:
[MIG] base_location: Migration to 15.0

OCA-git-bot committed:
partner_identification 14.0.1.3.0

victoralmau committed:
[FIX] base_location: address field sync city_id
TT33047

pedrocs-exo committed:
Translated using Weblate (Portuguese (Portugal))

Currently translated at 97.9% (48 of 49 strings)

Translation: partner-contact-14.0/partner-contact-14.0-partner_identification
Translate-URL: https://translation.odoo-community.org/projects/partner-contact-14-0/partner-contact-14-0-partner_identification/pt_PT/

victoralmau committed:
[FIX] base_location: Since odoo/odoo#81239 it is not necessary to add the zip_id field to _address_fields()
TT33047

pedrocs-exo committed:
Translated using Weblate (Portuguese)

Currently translated at 97.9% (48 of 49 strings)

Translation: partner-contact-14.0/partner-contact-14.0-partner_identification
Translate-URL: https://translation.odoo-community.org/projects/partner-contact-14-0/partner-contact-14-0-partner_identification/pt/

ValentinVinagre committed:
[FIX] base_location: Fix development_status

Commit:
[FIX] Fix access right issues when using actions

partner_pricelist_search 14.0.1.0.1

[IMP] update dotfiles [ci skip]

rbellanova committed:
Translated using Weblate (Italian)

Currently translated at 91.8% (45 of 49 strings)

Translation: partner-contact-14.0/partner-contact-14.0-partner_identification
Translate-URL: https://translation.odoo-community.org/projects/partner-contact-14-0/partner-contact-14-0-partner_identification/it/

rbellanova committed:
Translated using Weblate (Italian)

Currently translated at 91.8% (45 of 49 strings)

Translation: partner-contact-14.0/partner-contact-14.0-partner_identification
Translate-URL: https://translation.odoo-community.org/projects/partner-contact-14-0/partner-contact-14-0-partner_identification/it/

yankinmax committed:
[ADD] base_partner_company_group

vince-dynapps committed:
[MIG] partner_pricelist_search: Migration to 15.0

TT37297

pablontura committed:
Added translation using Weblate (Catalan)

Translated using Weblate (Catalan)

Currently translated at 100.0% (3 of 3 strings)

Translation: partner-contact-15.0/partner-contact-15.0-base_partner_company_group
Translate-URL: https://translation.odoo-community.org/projects/partner-contact-15-0/partner-contact-15-0-base_partner_company_group/ca/

pablontura committed:
Translated using Weblate (Catalan)

Currently translated at 88.8% (32 of 36 strings)

Translation: partner-contact-15.0/partner-contact-15.0-base_location
Translate-URL: https://translation.odoo-community.org/projects/partner-contact-15-0/partner-contact-15-0-base_location/ca/

yankinmax committed:
[IMP] partner_company_group: Add one2many counterpart for company_group_id

chienandalu committed:
[FIX] partner_pricelist_search: speedup search

TT39742

pedrobaeza committed:
[MIG] base_location: Migration to 16.0

- Use new _rec_names_search
- Use `name_get` instead of `display_name` field override.
- New dependency over `base_address_extended`.
- Remove useless test.
- Added <sheet> to city form

TT39871

Commit:
[UPD] Update base_location.pot

OCA-git-bot committed:
[UPD] README.rst

sergio-teruel committed:
[FIX] partner_pricelist_search: Remove list type also from search arguments

[UPD] Update partner_pricelist_search.pot

partner_pricelist_search 14.0.1.0.2

weblate committed:
Update translation files

Updated by "Update PO files to match POT (msgmerge)" hook in Weblate.

Translation: partner-contact-14.0/partner-contact-14.0-partner_pricelist_search
Translate-URL: https://translation.odoo-community.org/projects/partner-contact-14-0/partner-contact-14-0-partner_pricelist_search/

marylla committed:
Translated using Weblate (German)

Currently translated at 97.2% (35 of 36 strings)

Translation: partner-contact-16.0/partner-contact-16.0-base_location
Translate-URL: https://translation.odoo-community.org/projects/partner-contact-16-0/partner-contact-16-0-base_location/de/

marielejeune committed:
[MIG] partner_identification: Migration to 16.0

[UPD] Update partner_identification.pot

[UPD] README.rst

partner_identification 16.0.1.0.1

Update translation files

Updated by "Update PO files to match POT (msgmerge)" hook in Weblate.

Translation: partner-contact-16.0/partner-contact-16.0-partner_identification
Translate-URL: https://translation.odoo-community.org/projects/partner-contact-16-0/partner-contact-16-0-partner_identification/

Translated using Weblate (Italian)

Currently translated at 87.7% (43 of 49 strings)

Translation: partner-contact-16.0/partner-contact-16.0-partner_identification
Translate-URL: https://translation.odoo-community.org/projects/partner-contact-16-0/partner-contact-16-0-partner_identification/it/

Translated using Weblate (Italian)

Currently translated at 87.7% (43 of 49 strings)

Translation: partner-contact-16.0/partner-contact-16.0-partner_identification
Translate-URL: https://translation.odoo-community.org/projects/partner-contact-16-0/partner-contact-16-0-partner_identification/it/

Translated using Weblate (Croatian)

Currently translated at 30.6% (15 of 49 strings)

Translation: partner-contact-16.0/partner-contact-16.0-partner_identification
Translate-URL: https://translation.odoo-community.org/projects/partner-contact-16-0/partner-contact-16-0-partner_identification/hr/

Translated using Weblate (Croatian)

Currently translated at 51.0% (25 of 49 strings)

Translation: partner-contact-16.0/partner-contact-16.0-partner_identification
Translate-URL: https://translation.odoo-community.org/projects/partner-contact-16-0/partner-contact-16-0-partner_identification/hr/

Commit:
[UPD] Update base_location.pot

weblate committed:
Update translation files

Updated by "Update PO files to match POT (msgmerge)" hook in Weblate.

Translation: partner-contact-16.0/partner-contact-16.0-base_location
Translate-URL: https://translation.odoo-community.org/projects/partner-contact-16-0/partner-contact-16-0-base_location/

PierrickBrun committed:
Translated using Weblate (French)

Currently translated at 57.1% (20 of 35 strings)

Translation: partner-contact-16.0/partner-contact-16.0-base_location
Translate-URL: https://translation.odoo-community.org/projects/partner-contact-16-0/partner-contact-16-0-base_location/fr/

francesco-ooops committed:
Translated using Weblate (Italian)

Currently translated at 82.8% (29 of 35 strings)

Translation: partner-contact-16.0/partner-contact-16.0-base_location
Translate-URL: https://translation.odoo-community.org/projects/partner-contact-16-0/partner-contact-16-0-base_location/it/

badbole committed:
Translated using Weblate (Croatian)

Currently translated at 60.0% (21 of 35 strings)

Translation: partner-contact-16.0/partner-contact-16.0-base_location
Translate-URL: https://translation.odoo-community.org/projects/partner-contact-16-0/partner-contact-16-0-base_location/hr/

francesco-ooops committed:
Added translation using Weblate (Italian)

Translated using Weblate (Italian)

Currently translated at 50.0% (2 of 4 strings)

Translation: partner-contact-15.0/partner-contact-15.0-base_partner_company_group
Translate-URL: https://translation.odoo-community.org/projects/partner-contact-15-0/partner-contact-15-0-base_partner_company_group/it/

yankinmax committed:
[FIX] base_location: Waadt state already exists in Odoo

OCA-git-bot committed:
base_location 16.0.1.0.1

marielejeune committed:
[16.0][IMP] partner_identification: use odoo_test_helper in tests

Translated using Weblate (Italian)

Currently translated at 87.7% (43 of 49 strings)

Translation: partner-contact-16.0/partner-contact-16.0-partner_identification
Translate-URL: https://translation.odoo-community.org/projects/partner-contact-16-0/partner-contact-16-0-partner_identification/it/

Translated using Weblate (Slovenian)

Currently translated at 81.6% (40 of 49 strings)

Translation: partner-contact-16.0/partner-contact-16.0-partner_identification
Translate-URL: https://translation.odoo-community.org/projects/partner-contact-16-0/partner-contact-16-0-partner_identification/sl/

partner_identification 16.0.1.0.2

Translated using Weblate (Spanish)

Currently translated at 100.0% (49 of 49 strings)

Translation: partner-contact-16.0/partner-contact-16.0-partner_identification
Translate-URL: https://translation.odoo-community.org/projects/partner-contact-16-0/partner-contact-16-0-partner_identification/es/

Translated using Weblate (Portuguese (Brazil))

Currently translated at 48.9% (24 of 49 strings)

Translation: partner-contact-16.0/partner-contact-16.0-partner_identification
Translate-URL: https://translation.odoo-community.org/projects/partner-contact-16-0/partner-contact-16-0-partner_identification/pt_BR/

[UPD] README.rst

Update translation files

Updated by "Update PO files to match POT (msgmerge)" hook in Weblate.

Translation: partner-contact-16.0/partner-contact-16.0-partner_identification
Translate-URL: https://translation.odoo-community.org/projects/partner-contact-16-0/partner-contact-16-0-partner_identification/

francesco-ooops committed:
Translated using Weblate (Italian)

Currently translated at 82.8% (29 of 35 strings)

Translation: partner-contact-16.0/partner-contact-16.0-base_location
Translate-URL: https://translation.odoo-community.org/projects/partner-contact-16-0/partner-contact-16-0-base_location/it/

sysadminmatmoz committed:
Translated using Weblate (Slovenian)

Currently translated at 48.5% (17 of 35 strings)

Translation: partner-contact-16.0/partner-contact-16.0-base_location
Translate-URL: https://translation.odoo-community.org/projects/partner-contact-16-0/partner-contact-16-0-base_location/sl/

CedricAtanane committed:
Added translation using Weblate (French)

Added translation using Weblate (French (France) (fr_FR))

Translated using Weblate (French)

Currently translated at 100.0% (4 of 4 strings)

Translation: partner-contact-15.0/partner-contact-15.0-base_partner_company_group
Translate-URL: https://translation.odoo-community.org/projects/partner-contact-15-0/partner-contact-15-0-base_partner_company_group/fr/

Translated using Weblate (French (France) (fr_FR))

Currently translated at 100.0% (4 of 4 strings)

Translation: partner-contact-15.0/partner-contact-15.0-base_partner_company_group
Translate-URL: https://translation.odoo-community.org/projects/partner-contact-15-0/partner-contact-15-0-base_partner_company_group/fr_FR/

Highcooley committed:
[MIG] base_partner_company_group: Migration to 16.0

Highcooley committed:
[MIG] partner_pricelist_search: Migration to 16.0

Highcooley committed:
base_partner_company_group: fix author

Co-authored-by: Simone Orsi <simahawk@users.noreply.github.com>

victor-champonnois committed:
Translated using Weblate (French)

Currently translated at 100.0% (35 of 35 strings)

Translation: partner-contact-16.0/partner-contact-16.0-base_location
Translate-URL: https://translation.odoo-community.org/projects/partner-contact-16-0/partner-contact-16-0-base_location/fr/

badbole committed:
Translated using Weblate (Croatian)

Currently translated at 68.5% (24 of 35 strings)

Translation: partner-contact-16.0/partner-contact-16.0-base_location
Translate-URL: https://translation.odoo-community.org/projects/partner-contact-16-0/partner-contact-16-0-base_location/hr/

badbole committed:
Added translation using Weblate (Croatian)

weblate committed:
Update translation files

Updated by "Update PO files to match POT (msgmerge)" hook in Weblate.

Translation: partner-contact-16.0/partner-contact-16.0-partner_pricelist_search
Translate-URL: https://translation.odoo-community.org/projects/partner-contact-16-0/partner-contact-16-0-partner_pricelist_search/

adrianojprado committed:
Added translation using Weblate (Portuguese (Brazil))

Ivorra78 committed:
Translated using Weblate (Spanish)

Currently translated at 100.0% (4 of 4 strings)

Translation: partner-contact-16.0/partner-contact-16.0-base_partner_company_group
Translate-URL: https://translation.odoo-community.org/projects/partner-contact-16-0/partner-contact-16-0-base_partner_company_group/es/

OCA-git-bot committed:
[UPD] README.rst

OCA-git-bot committed:
[UPD] README.rst

gva-acsone committed:
Added translation using Weblate (French (Belgium))

weblate committed:
Update translation files

Updated by "Update PO files to match POT (msgmerge)" hook in Weblate.

Translation: partner-contact-16.0/partner-contact-16.0-base_partner_company_group
Translate-URL: https://translation.odoo-community.org/projects/partner-contact-16-0/partner-contact-16-0-base_partner_company_group/

weblate committed:
Update translation files

Updated by "Update PO files to match POT (msgmerge)" hook in Weblate.

Translation: partner-contact-16.0/partner-contact-16.0-base_location
Translate-URL: https://translation.odoo-community.org/projects/partner-contact-16-0/partner-contact-16-0-base_location/

JordiMForgeFlow committed:
[IMP] partner_identification*: move _search_duplicate

OCA-git-bot committed:
[BOT] post-merge updates

josep-tecnativa committed:
[IMP] base_partner_company_group: test performance improvement
- Include context keys for avoiding mail operations overhead.

adrianojprado committed:
Translated using Weblate (Portuguese (Brazil))

Currently translated at 100.0% (35 of 35 strings)

Translation: partner-contact-16.0/partner-contact-16.0-base_location
Translate-URL: https://translation.odoo-community.org/projects/partner-contact-16-0/partner-contact-16-0-base_location/pt_BR/

NICO-SOLUTIONS committed:
[IMP] partner_identification: pre-commit auto fixes

NICO-SOLUTIONS committed:
[MIG] partner_identification: Migration to 17.0

OCA-git-bot committed:
[BOT] post-merge updates

Commit:
[UPD] Update partner_identification.pot

OCA-git-bot committed:
[BOT] post-merge updates

mymage committed:
Translated using Weblate (Italian)

Currently translated at 100.0% (4 of 4 strings)

Translation: partner-contact-16.0/partner-contact-16.0-base_partner_company_group
Translate-URL: https://translation.odoo-community.org/projects/partner-contact-16-0/partner-contact-16-0-base_partner_company_group/it/

mymage committed:
Added translation using Weblate (Italian)

mymage committed:
Translated using Weblate (Italian)

Currently translated at 100.0% (4 of 4 strings)

Translation: partner-contact-16.0/partner-contact-16.0-partner_pricelist_search
Translate-URL: https://translation.odoo-community.org/projects/partner-contact-16-0/partner-contact-16-0-partner_pricelist_search/it/

mymage committed:
Translated using Weblate (Italian)

Currently translated at 100.0% (35 of 35 strings)

Translation: partner-contact-16.0/partner-contact-16.0-base_location
Translate-URL: https://translation.odoo-community.org/projects/partner-contact-16-0/partner-contact-16-0-base_location/it/

Ivorra78 committed:
Added translation using Weblate (Spanish)

Ivorra78 committed:
Translated using Weblate (Spanish)

Currently translated at 100.0% (9 of 9 strings)

Translation: partner-contact-14.0/partner-contact-14.0-partner_identification_unique_by_category
Translate-URL: https://translation.odoo-community.org/projects/partner-contact-14-0/partner-contact-14-0-partner_identification_unique_by_category/es/

payen000 committed:
[FIX] partner_firstname: Avoid name fields rewrite on partner creation

Previously, creating a non-company partner by passing the `name` field
would trigger the `_inverse_name_after_cleaning_whitespace` method which,
for values of `firstname` and `lastname` with multiple words, would result
in most of those words being set to the `lastname` field, even if the
correct values were explicitly passed to the `create` method. For example,
duplicating a non-company partner with the following field values:

firstname = Francisco Javier
lastname = Garcia Cabeza De Vaca

Would result in a partner with these values:

firstname = Francisco
lastname = Javier Garcia Cabeza De Vaca (copy)

A similar issue can be reproduced when recalculating names: when
explicitly calling `_compute_name`, the
`_inverse_name_after_cleaning_whitespace` is called and the `firstname`
and `lastname` fields are modified, if they contain more than one word.

This commit solves the issue and prevents the `_inverse_name_after_cleaning_whitespace`
from being called when creating a non-company partner while explicitly
providing the `firstname` or `lastname` fields, and when recalculating
names.

In addition to the 16.0: Complying with E501 linting rule

payen000 committed:
[FIX] partner_second_lastname: Ensure '(copy)' suffix on partner duplication

This PR ensures that when duplicating a partner, the '(copy)' suffix is
added at the end of `lastname2` instead of `lastname`, in case the order
from settings is `first_last`; otherwise, no action is needed.

Commit:
[IMP] base_location: pre-commit stuff

Commit:
[MIG] base_location: Migration to 17.0

weblate committed:
Update translation files

Updated by "Update PO files to match POT (msgmerge)" hook in Weblate.

Translation: partner-contact-17.0/partner-contact-17.0-partner_identification
Translate-URL: https://translation.odoo-community.org/projects/partner-contact-17-0/partner-contact-17-0-partner_identification/

mymage committed:
Translated using Weblate (Italian)

Currently translated at 100.0% (16 of 16 strings)

Translation: partner-contact-17.0/partner-contact-17.0-partner_firstname
Translate-URL: https://translation.odoo-community.org/projects/partner-contact-17-0/partner-contact-17-0-partner_firstname/it/

mymage committed:
Translated using Weblate (Italian)

Currently translated at 100.0% (48 of 48 strings)

Translation: partner-contact-17.0/partner-contact-17.0-partner_identification
Translate-URL: https://translation.odoo-community.org/projects/partner-contact-17-0/partner-contact-17-0-partner_identification/it/

mymage committed:
Added translation using Weblate (Italian)

Commit:
[UPD] Update base_location.pot

OCA-git-bot committed:
[BOT] post-merge updates

weblate committed:
Update translation files

Updated by "Update PO files to match POT (msgmerge)" hook in Weblate.

Translation: partner-contact-17.0/partner-contact-17.0-base_location
Translate-URL: https://translation.odoo-community.org/projects/partner-contact-17-0/partner-contact-17-0-base_location/

BurkhalterY committed:
[FIX] partner_firstname: fix error when creating a user from an employee

Steps to reproduce:
- Install `hr` and `partner_firstname`
- Go to Employees > any employee > action menu > Create User > Save
- 💥 Traceback

Suspected cause: in `defaul_get()`, the field `name` is not present in `fields_list` (probably because it's computed). So `result.get("name", "")` will always return `""` and `firstname` and `lastname` are not computed.

nguyenminhchien committed:
[IMP] partner_pricelist_search: pre-commit auto fixes

nguyenminhchien committed:
[MIG] partner_pricelist_search: Migration to 17.0

OCA-git-bot committed:
[BOT] post-merge updates

SirAionTech committed:
[FIX] partner_firstname: Multiple creation with "name"
Otherwise the following is raised
> odoo.addons.partner_firstname.exceptions.EmptyNamesError: ("Error(s) with partner 1425's name.", 'No name is set.')

Commit:
[IMP] base_partner_company_group: pre-commit auto fixes

Commit:
[MIG] base_partner_company_group: Migration to 17.0

Commit:
[UPD] Update base_partner_company_group.pot

OCA-git-bot committed:
[BOT] post-merge updates

jakobkrabbe committed:
Translated using Weblate (Swedish)

Currently translated at 100.0% (35 of 35 strings)

Translation: partner-contact-17.0/partner-contact-17.0-base_location
Translate-URL: https://translation.odoo-community.org/projects/partner-contact-17-0/partner-contact-17-0-base_location/sv/

jakobkrabbe committed:
Translated using Weblate (Swedish)

Currently translated at 100.0% (16 of 16 strings)

Translation: partner-contact-17.0/partner-contact-17.0-partner_firstname
Translate-URL: https://translation.odoo-community.org/projects/partner-contact-17-0/partner-contact-17-0-partner_firstname/sv/

jakobkrabbe committed:
Translated using Weblate (Swedish)

Currently translated at 100.0% (48 of 48 strings)

Translation: partner-contact-17.0/partner-contact-17.0-partner_identification
Translate-URL: https://translation.odoo-community.org/projects/partner-contact-17-0/partner-contact-17-0-partner_identification/sv/

Commit:
[UPD] Update partner_firstname.pot

OCA-git-bot committed:
[BOT] post-merge updates

weblate committed:
Update translation files

Updated by "Update PO files to match POT (msgmerge)" hook in Weblate.

Translation: partner-contact-17.0/partner-contact-17.0-partner_firstname
Translate-URL: https://translation.odoo-community.org/projects/partner-contact-17-0/partner-contact-17-0-partner_firstname/

mymage committed:
Translated using Weblate (Italian)

Currently translated at 100.0% (18 of 18 strings)

Translation: partner-contact-17.0/partner-contact-17.0-partner_firstname
Translate-URL: https://translation.odoo-community.org/projects/partner-contact-17-0/partner-contact-17-0-partner_firstname/it/

rpinset committed:
[FIX] partner_firstname : bug "create and edit" from invoice and sale order #1757 #1725

OCA-git-bot committed:
[BOT] post-merge updates

Commit:
[IMP] partner_contact_role: pre-commit execution

Commit:
[IMP] partner_contact_role: pre-commit execution

Commit:
[IMP] partner_contact_role: pre-commit execution

Commit:
[IMP] partner_contact_role: pre-commit execution

esteveblanch committed:
Translated using Weblate (Catalan)

Currently translated at 100.0% (18 of 18 strings)

Translation: partner-contact-17.0/partner-contact-17.0-partner_firstname
Translate-URL: https://translation.odoo-community.org/projects/partner-contact-17-0/partner-contact-17-0-partner_firstname/ca/

esteveblanch committed:
Translated using Weblate (Catalan)

Currently translated at 100.0% (35 of 35 strings)

Translation: partner-contact-17.0/partner-contact-17.0-base_location
Translate-URL: https://translation.odoo-community.org/projects/partner-contact-17-0/partner-contact-17-0-base_location/ca/

chaule97 committed:
[MIG] partner_identification: Migration to 18.0

chaule97 committed:
[IMP] partner_identification: pre-commit auto fixes

chaule97 committed:
[IMP] partner_identification_unique_by_category: pre-commit auto fixes

chaule97 committed:
[MIG] partner_identification_unique_by_category: Migration to 18.0

xaviedoanhduy committed:
[MIG] partner_pricelist_search: Migration to 18.0

chaule97 committed:
Add test-requirements.txt

nguyenminhchien committed:
[MIG] base_partner_company_group: Migration to 18.0

Kimkhoi3010 committed:
[MIG] base_partner_sequence: Migration to 18.0

Kimkhoi3010 committed:
oca-port: blacklist PR(s) , 1472 for base_location

Kimkhoi3010 committed:
[IMP] base_location: pre-commit auto fixes

Kimkhoi3010 committed:
[MIG] base_location: Migration to 18.0

Commit:
[IMP] partner_firstname: pre-commit stuff

Commit:
[MIG] partner_firstname: Migration to 18.0

chaule97 commented on base_location/models/res_partner.py:
```diff
@@ -0,0 +1,168 @@
+            error_dict = {"partner": rec.name, "location": rec.zip_id.name}
+            if rec.zip_id.city_id.country_id != rec.country_id:
+                raise ValidationError(
+                    _(
+                        "The country of the partner %(partner)s differs from that in "
+                        "location %(location)s"
+                    )
+                    % error_dict
```
```suggestion
                   self.env._(
                        "The country of the partner %(partner)s differs from that in "
                        "location %(location)s",
                        error_dict
                    )
```

chaule97 commented on base_location/models/res_partner.py:
```diff
@@ -0,0 +1,168 @@
+                )
+            if rec.zip_id.city_id.state_id != rec.state_id:
+                raise ValidationError(
+                    _(
+                        "The state of the partner %(partner)s differs from that in "
+                        "location %(location)s"
+                    )
+                    % error_dict
```
```suggestion
                    self.env._(
                        "The state of the partner %(partner)s differs from that in "
                        "location %(location)s",
                        error_dict
                    )
```

chaule97 commented on base_location/models/res_partner.py:
```diff
@@ -0,0 +1,168 @@
+                )
+            if rec.zip_id.city_id != rec.city_id:
+                raise ValidationError(
+                    _(
+                        "The city of the partner %(partner)s differs from that in "
+                        "location %(location)s"
+                    )
+                    % error_dict
```
same above

chaule97 commented on base_location/models/res_partner.py:
```diff
@@ -0,0 +1,168 @@
+                )
+            if rec.zip_id.name != rec.zip:
+                raise ValidationError(
+                    _(
+                        "The zip of the partner %(partner)s differs from that in "
+                        "location %(location)s"
+                    )
+                    % error_dict
```
same above

chaule97 commented on base_location/models/res_partner.py:
```diff
@@ -0,0 +1,168 @@
+
+from lxml import etree
+
+from odoo import _, api, fields, models
```
```suggestion
from odoo import api, fields, models
```
https://github.com/odoo/odoo/commit/b794f0f332f473deb2c04eba60baf4761db3b508

chaule97 commented on base_location/models/res_partner.py:
```diff
@@ -0,0 +1,168 @@
+            error_dict = {"partner": rec.name, "location": rec.zip_id.name}
+            if rec.zip_id.city_id.country_id != rec.country_id:
+                raise ValidationError(
+                    self.env._(
+                        "The country of the partner %(partner)s differs from that in "
+                        "location %(location)s"
+                    )
+                    % error_dict
```
```suggestion
                    self.env._(
                        "The country of the partner %(partner)s differs from that in "
                        "location %(location)s",
                        **error_dict,
                    )
```

chaule97 commented on base_location/models/res_partner.py:
```diff
@@ -0,0 +1,168 @@
+                )
+            if rec.zip_id.city_id.state_id != rec.state_id:
+                raise ValidationError(
+                    self.env._(
+                        "The state of the partner %(partner)s differs from that in "
+                        "location %(location)s"
+                    )
+                    % error_dict
```
```suggestion
                    self.env._(
                        "The state of the partner %(partner)s differs from that in "
                        "location %(location)s",
                        **error_dict,
                    )
```

chaule97 commented on base_location/models/res_partner.py:
```diff
@@ -0,0 +1,168 @@
+                )
+            if rec.zip_id.city_id != rec.city_id:
+                raise ValidationError(
+                    self.env._(
+                        "The city of the partner %(partner)s differs from that in "
+                        "location %(location)s"
+                    )
+                    % error_dict
```
```suggestion
                    self.env._(
                        "The city of the partner %(partner)s differs from that in "
                        "location %(location)s",
                        **error_dict,
                    )
```

chaule97 commented on base_location/models/res_partner.py:
```diff
@@ -0,0 +1,168 @@
+                )
+            if rec.zip_id.name != rec.zip:
+                raise ValidationError(
+                    self.env._(
+                        "The zip of the partner %(partner)s differs from that in "
+                        "location %(location)s"
+                    )
+                    % error_dict
```
```suggestion
                    self.env._(
                        "The zip of the partner %(partner)s differs from that in "
                        "location %(location)s",
                        **error_dict,
                    )
```

OCA-git-bot committed:
Merge PR #1860 into 18.0

Signed-off-by pedrobaeza

Commit:
[UPD] Update partner_ref_unique.pot

OCA-git-bot committed:
[BOT] post-merge updates

weblate committed:
Update translation files

Updated by "Update PO files to match POT (msgmerge)" hook in Weblate.

Translation: partner-contact-18.0/partner-contact-18.0-partner_ref_unique
Translate-URL: https://translation.odoo-community.org/projects/partner-contact-18-0/partner-contact-18-0-partner_ref_unique/

OCA-git-bot committed:
Merge PR #1861 into 18.0

Signed-off-by pedrobaeza

Commit:
[UPD] Update partner_tz.pot

OCA-git-bot committed:
[BOT] post-merge updates

OCA-git-bot committed:
[IMP] update dotfiles

OCA-git-bot committed:
Merge PR #1866 into 18.0

Signed-off-by pedrobaeza

Commit:
[UPD] Update partner_pricelist_search.pot

OCA-git-bot committed:
[BOT] post-merge updates

OCA-git-bot committed:
Merge PR #1865 into 18.0

Signed-off-by rousseldenis

Commit:
[UPD] Update partner_firstname.pot

OCA-git-bot committed:
[BOT] post-merge updates

weblate committed:
Update translation files

Updated by "Update PO files to match POT (msgmerge)" hook in Weblate.

Translation: partner-contact-18.0/partner-contact-18.0-partner_firstname
Translate-URL: https://translation.odoo-community.org/projects/partner-contact-18-0/partner-contact-18-0-partner_firstname/

OCA-git-bot committed:
Merge PR #1858 into 18.0

Signed-off-by rousseldenis

Commit:
[UPD] Update partner_identification.pot

OCA-git-bot committed:
[BOT] post-merge updates

weblate committed:
Update translation files

Updated by "Update PO files to match POT (msgmerge)" hook in Weblate.

Translation: partner-contact-18.0/partner-contact-18.0-partner_identification
Translate-URL: https://translation.odoo-community.org/projects/partner-contact-18-0/partner-contact-18-0-partner_identification/

OCA-git-bot committed:
Merge PR #1863 into 18.0

Signed-off-by simahawk

Commit:
[UPD] Update base_partner_sequence.pot

OCA-git-bot committed:
[BOT] post-merge updates

OCA-git-bot committed:
Merge PR #1871 into 18.0

Signed-off-by simahawk

Commit:
[UPD] Update base_partner_company_group.pot

OCA-git-bot committed:
[BOT] post-merge updates

weblate committed:
Update translation files

Updated by "Update PO files to match POT (msgmerge)" hook in Weblate.

Translation: partner-contact-18.0/partner-contact-18.0-base_partner_company_group
Translate-URL: https://translation.odoo-community.org/projects/partner-contact-18-0/partner-contact-18-0-base_partner_company_group/

weblate committed:
Update translation files

Updated by "Update PO files to match POT (msgmerge)" hook in Weblate.

Translation: partner-contact-18.0/partner-contact-18.0-base_partner_sequence
Translate-URL: https://translation.odoo-community.org/projects/partner-contact-18-0/partner-contact-18-0-base_partner_sequence/

mymage committed:
Translated using Weblate (Italian)

Currently translated at 100.0% (77 of 77 strings)

Translation: partner-contact-18.0/partner-contact-18.0-partner_identification
Translate-URL: https://translation.odoo-community.org/projects/partner-contact-18-0/partner-contact-18-0-partner_identification/it/

OCA-git-bot committed:
Merge PR #1859 into 18.0

Signed-off-by simahawk

Commit:
[UPD] Update partner_identification_unique_by_category.pot

OCA-git-bot committed:
[BOT] post-merge updates

weblate committed:
Update translation files

Updated by "Update PO files to match POT (msgmerge)" hook in Weblate.

Translation: partner-contact-18.0/partner-contact-18.0-partner_identification_unique_by_category
Translate-URL: https://translation.odoo-community.org/projects/partner-contact-18-0/partner-contact-18-0-partner_identification_unique_by_category/

mymage committed:
Translated using Weblate (Italian)

Currently translated at 100.0% (6 of 6 strings)

Translation: partner-contact-18.0/partner-contact-18.0-partner_identification_unique_by_category
Translate-URL: https://translation.odoo-community.org/projects/partner-contact-18-0/partner-contact-18-0-partner_identification_unique_by_category/it/
